config.go

```diff
@@ -12,23 +14,54 @@ type Config struct {
 	Packages map[string]Package `yaml:"packages"`
 }
 
+// orderedConfig is used internally to check for alphabetical ordering in the YAML
+// configuration. A yaml.MapSlice perservers ordering of keys: https://godoc.org/gopkg.in/yaml.v2#MapSlice
+type orderedConfig struct {
+	URL      string        `yaml:"url"`
+	Packages yaml.MapSlice `yaml:"packages"`
+}
+
 // Package details the options available for each repo
 type Package struct {
 	Repo string `yaml:"repo"`
 }
 
+// ensureAlphabetical checks that the packages are listed alphabetically in the configuration.
+func ensureAlphabetical(data []byte) bool {
+	var c orderedConfig
+
+	if err := yaml.Unmarshal(data, &c); err != nil {
+		return false
+	}
+
+	packageNames := make([]string, len(c.Packages))
+	for _, v := range c.Packages {
+		name, ok := v.Key.(string)
+		if !ok {
+			return false
+		}
+		packageNames = append(packageNames, name)
```

Comment:
You're creating a slice of length `len(c.Packages)` and appending to it.  This
will end up with that many empty strings at the start of the slice followed by
the names. Example: https://play.golang.org/p/rvWTmSpP9Q

You want to use `make([]string, 0, len(c.Packages))` if you want to append to
the slice.

    packageNames := make([]string, 0, len(c.Packages))
    for _, v := range c.Packages {
        // ..
        packageNames = append(packageNames, name)
    }

Alternatively, you can assign to the indexes:

    packageNames := make([]string, len(c.Packages))
    for i, v := range c.Packages {
        // ...
        packageNames[i] = name
    }